core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -451,13 +454,16 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
         m_vector_size(arg_policy.impl_vector_length()) {
     // FIXME_SYCL optimize
     if (m_team_size < 0) m_team_size = 32;
+    // choose team_size as the next multiple of m_vector_size
+    m_team_size =
+        ((m_team_size + m_vector_size - 1) / m_vector_size) * m_vector_size;
```

Comment:
This feels iffy. You divide vector length away further up anyway, and using this team_size for the scratch calculation without dividing vector length away again below is probably wrong or? I mean since you force vectorlength 1 it doesn't fail you but we should have the right expression here anyway.